Formula/o/orbuculum.rb

```diff
@@ -0,0 +1,77 @@
+class Orbuculum < Formula
+  desc "Arm Cortex-M SWO/SWV Demux and Postprocess"
+  homepage "https://github.com/orbcode/orbuculum"
+  url "https://github.com/orbcode/orbuculum/archive/refs/tags/V2.1.0.tar.gz"
+  sha256 "ccdd86130094001a0ab61e5501a6636e12c82b0b44690795a2911c65c5618c46"
+  license "BSD-3-Clause"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "capstone"
+  depends_on "libusb"
+  depends_on "sdl2"
+  depends_on "zeromq"
+  uses_from_macos "ncurses"
+
+  on_macos do
+    depends_on "libelf"
+  end
+  on_linux do
+    depends_on "elfutils"
+  end
+
+  def install
+    system "meson", "setup", "build", *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build", "--tags", "devel,runtime"
+  end
+
+  # The tools do not report their version correctly when installed from a
+  # GitHub release rather than from Git directly for versions <= 2.1.0.
+  # This has now been rectified, and the versions can be tested for future
+  # releases.
+  test do
+    shell_output("#{bin}/orbuculum --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbcat --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbdump --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbfifo --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orblcd --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbmortem --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbprofile --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbstat --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbtop --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbtrace --version", 255)
+  end
+
+  test do
+    shell_output("#{bin}/orbzmq --version", 255)
+  end
```

Comment:
This should all be a single test block, and we need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Tests all put into a single block in 13ca97e23c06d4f12bb043362128f7f3965b5112.

Re. the tests themselves, I think it will be hard to write very meaningful tests as the software requires hardware to be connected to the host machine. I see this is similar to the last two cases given:
> * If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
> * Same if the software requires a virtual machine, docker instance, etc. to be running.

If you run any of the tools without the hardware connected, their behaviour is to wait for the hardware to be connected — how would I go about testing this?

Reply:
Does it wait indefinitely? Does it print anything? If so, you can start it, wait for output and kill it again.

Reply:
Most of the tools wait indefinitely with no output until hardware is connected.

Some of them require debug information via an ELF file, and if this is not provided they immediately fail. I have updated to test this behaviour in b52ca246fce16cae896b4d9138b9002aa94950bc.